sklearn/tests/test_discriminant_analysis.py

```diff
@@ -296,6 +297,31 @@ def test_lda_dimension_warning(n_classes, n_features):
         assert_warns_message(FutureWarning, future_msg, lda.fit, X, y)
 
 
+@pytest.mark.parametrize("data_type, expected_type", [
+    (np.float32, np.float32),
+    (np.float64, np.float64),
+    (np.int32, np.float64),
```

Comment:
Going from int32 to float64 is not what was done by isotonic regression isofused branch this morning. We should be consistent here.

Reply:
I am +1 for int32 -> float64. It was the semantic for `check_array`.
@ogrisel was also for this conversion. I would think that we should correct the isotonic regression then.

Reply:
+1. Merging. This is a net improvement. We'll nitpick later :D